hollow/src/main/java/com/netflix/hollow/api/consumer/HollowConsumer.java

```diff
@@ -405,15 +409,16 @@ public long getToVersion() {
      */
     public static interface AnnouncementWatcher {
 
-        public static final long NO_ANNOUNCEMENT_AVAILABLE = Long.MIN_VALUE;
+        public static final long NO_VERSION_AVAILABLE = Long.MIN_VALUE;
```

Comment:
I wonder if we can keep the *announcement* nomenclature?

For cloud environment, we use two topics: main topic and the pin topic. In both cases we "announce" a version to that topic. Though maybe it's more strained to talk in terms of "announcing a pinned version" versus talking about "announcing" and "pinning" as two separate verbs. Or maybe the verb "pinning" is shorthand for "announcing a pinned version" and that's equally clear.

Reply:
I literally spent a long time going on circles on this. I think you could announce two type of versions, the "pinned" one and the "normal"? (don't know how to call it) and `NO_ANNOUNCEMENT_AVAILABLE` could prevail. 

To be honest, while doing these changes, I thought `NO_ANNOUNCEMENT_AVAILABLE` wasn't the proper one because I'm not used to see the pinning as part of the Hollow internals, so you always think of announcement as a new version that came in. I think is more of semantics of announcement on a daily-basis usage rather than if it fits or not... hope I'm being clear on this. Now it makes more sense to keep it as it is.

Also, I think `NO_ANNOUNCEMENT_AVAILABLE` could be part of some sort of constants class that could be shared across `VersionPinner`, `HollowProducer` and `HollowConsumer`; in particular for the Filesystem implementations.

 Overall, I think pinning needs more exposure, either at docs level or as part of the public API so people could adopt easily that pattern which IMO is very useful. In my experience, I talked to other teams that didn't think about it and didn't read the [Pinning Consumer](http://hollow.how/infrastructure/#pinning-consumers) section and this could help them while triaging issues

Reply:
> so you always think of announcement as a new version that came in. I think is more of semantics of announcement on a daily-basis usage

That's a good insight. Even if you can learn that Hollow treats pinning like a flavor of announcing, is that a natural way of thinking about it or will you always find yourself going "oh yeah, pins are announcements, too". It's one thing if that happens to an implementation detail and another if it's the metaphor in the public API.

> Overall, I think pinning needs more exposure, either at docs level or as part of the public API so people could adopt easily that pattern which IMO is very useful.

👍

Reply:
I like the idea. Announcement naturally means a new version is available, whether it's pinned version or a new version that producer published.